### PR TITLE
chore(sdk): clean up references to `dsl.Condition`

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
- Cleans up references to `dsl.Condition` in test code, except for a few instances to ensure backward compatibility is preserved
- Ensures error messages present the form of `dsl.Condition`/`dsl.If`/etc. actually used by users via `group.__class__.__name__`

**Checklist:**
- [x] The title for your pull request (PR) should follow our title 
convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/compiler/compiler_test.py [hunk 1]
```diff
@@ -524,12 +524,12 @@ def dummy_op(msg: str = ''):
 
         with self.assertRaisesRegex(
                 compiler_utils.InvalidTopologyException,
-                r'Illegal task dependency across DSL context managers\. A downstream task cannot depend on an upstream task within a dsl\.Condition context unless the downstream is within that context too\. Found task dummy-op which depends on upstream task producer-op within an uncommon dsl\.Condition context\.'
+                r'Illegal task dependency across DSL context managers\. A downstream task cannot depend on an upstream task within a dsl\.If context unless the downstream is within that context too\. Found task dummy-op which depends on upstream task producer-op within an uncommon dsl\.If context\.'
         ):
 
             @dsl.pipeline(name='test-pipeline')
             def my_pipeline(val: bool):
-                with dsl.Condition(val == False):
```
Comment (chensun) [inline sdk/python/kfp/compiler/compiler_test.py:532]: Given `dsl.Condition` is not deprecated yet, we need to ensure it still functions. From this perspective, we should keep its references in test.

File: sdk/python/test_data/pipelines/pipeline_with_condition.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def my_pipeline(text: str = 'condition test'):
     flip1 = flip_coin_op().set_caching_options(False)
     print_op(msg=flip1.output)
 
-    with dsl.Condition(flip1.output == 'heads'):
```
Comment (chensun) [inline sdk/python/test_data/pipelines/pipeline_with_condition.py:39]: Would there be some duplication between these the if/elif/else samples we might already have? Consider reducing the duplication.

File: sdk/python/kfp/compiler/compiler_test.py (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/python/kfp/compiler/compiler_test.py:532]: Closed this PR.

**Pull request closed without merging**